core/test-runner.ts

```diff
@@ -18,6 +19,16 @@ export class TestRunner {
    private _testResults = new TestSetResults();
    private _currentTestResults: TestResults;
    private _currentTestFixtureResults: TestFixtureResults;
+   private _output: TestOutput;
+
+   constructor (output?: TestOutput) {
+       // If we were given a TestOutput, use it, otherwise make one
+       if (output !== undefined) {
```

Comment:
Exactly, if they want to make a TestRunner they can just do `new TestRunner` and pass in their own `TestOutcome` class  if they want